contracts/base/LoansBase.sol

```diff
@@ -494,10 +494,57 @@ contract LoansBase is LoansInterface, Base {
         address payable recipient
     ) internal {
         if (loans[loanID].escrow != address(0x0)) {
-            EscrowInterface(loans[loanID].escrow).claimTokens(recipient);
+            /**
+                - Check the value of the liquidator will get as a reward - total amount owed * 1.05 and converted into collateral asset 
+                settings().chainlinkAggregator().valueFor(
+                    baseAddress,
+                    quoteAddress,
+                    baseAmount
+                ); */
+            uint256 totalOwed = _getTotalOwed(loanID);
+            uint256 amountToLiquidatorInLendingTokens = totalOwed.mul(105).div(100);
+            uint256 amountToLiquidatorInCollateralTokens = settings()
+                .chainlinkAggregator()
+                .valueFor(
+                lendingPool.lendingToken(),
+                collateralToken,
+                amountToLiquidatorInLendingTokens
+            );
+            uint256 leftoverTokens;
+            /** - if converted collateral value > amount 
+                    _payOutCollateral(loanID, amount, recipient);
+                    - amount of value in tokens from the escrow to be claimed = converted collateral value - amount [if escrow exists], claimTokens(receipient, converted collateral value) - else the borrower has to call the claimTokens */
+            if (amountToLiquidatorInCollateralTokens > totalOwed) {
+                _payOutCollateral(loanID, amount, recipient);
+                leftoverTokens = amountToLiquidatorInCollateralTokens - totalOwed;
+                EscrowInterface(loans[loanID].escrow).claimTokensByCollateralValue(
+                    loans[loanID].loanTerms.borrower,
+                    leftoverTokens
+                );
+            } else if (amountToLiquidatorInCollateralTokens == totalOwed) {
+                /** - if converted collateral value = amount
+                    _payOutCollateral(loanID, amount, recipient); */
+                _payOutCollateral(loanID, amount, recipient);
```

Comment:
You are comparing collateral token value to lending token value here. Instead you should compare `amountToLiquidatorInCollateralTokens >= loans[loanID].collateral` to see if we need to dip into the escrow tokens to collect the remaining reward.

1. Payout the full collateral amount to the liquidator.
1. If `remaining value > 0` 
    1. Call escrow to claim the remaining value to go to the liquidator.
1. Do not claim any token value for the borrower, as the liquidator will end up paying the gas.

Reply:
Good catch, changed